specs/LaunchingRegisteredProtocols.md

```diff
@@ -0,0 +1,305 @@
+# Background
+
+We are exposing an event that will be raised when an attempt to launch an external protocol is made. The host will be given the option to cancel the launch, handle the popup dialog to hide the dialog, as well as disable the checkbox that, if selected, will give permissions to always allow the launch.
+
+# Description
+
+This event will be raised before the external protocol launch occurs. Currently a popup dialog is displayed in which the user can click `Open` or `Cancel`. If the request is made from a [trustworthy origin](#https://w3c.github.io/webappsec-secure-contexts/#potentially-trustworthy-origin) a checkmark box will be displayed that will allow the user to always allow this external protocol from this origin. The `NavigationStarting`, `NavigationCompleted`, `SourceChanged`, `ContentLoading`, and `HistoryChanged` events will not be raised when a request is made to launch an external protocol.
+
+The `LaunchingExternalProtocol` event will be raised on either `CoreWebView2` or `CoreWebView2Frame` depending on if the launch request is originating from the main frame or a non-main frame. A current limitation of our `CoreWebView2Frame` API is that it only supports top level iframes. Any nested iframes will not have a `CoreWebView2Frame` associated with them. In the case of a nested iframe requesting the external protocol launch, the event will be raised from the top level iframe.
```

Comment:
Raising the event on the CoreWebView2Frame for a grand child iframe is a little weird. It doesn't act that way for CoreWebView2's event. If we add the ability to get grand child iframes as CoreWebView2Frames in the future we'd be sort of stuck with this weird behavior.

Would it work to have just one CoreWebView2.LaunchingExternalProtocol event that is raised for top level documents and iframes?

Reply:
Does this mean having one event that would be raised for both, with a boolean identifying if this is an iframe or top level document?